apps/aws-notebook/samples/ecs/fargate/vpn/main.tf

```diff
@@ -0,0 +1,252 @@
+  container_definitions = jsonencode([
+    {
+      name      = "ssh-server"
+      image     = "926120211684.dkr.ecr.us-west-1.amazonaws.com/dev/tzz/ssh-server:latest"
+      essential = true
+      portMappings = [
+        {
+          containerPort = 2222
+          hostPort      = 2222
+        }
+      ]
+    }
+  ])
+}
+
+## Create ECS service
+resource "aws_ecs_service" "dev-connection-ecs-fargate-vpn" {
+  launch_type         = "FARGATE"
+  task_definition     = aws_ecs_task_definition.dev-connection-ecs-fargate-vpn.arn
+  name                = "dev-connection-ecs-fargate-vpn-ssh-server"
+  cluster             = aws_ecs_cluster.dev-connection-ecs-fargate-vpn.id
+  scheduling_strategy = "REPLICA"
+  desired_count       = 1
+  network_configuration {
+    subnets          = [aws_subnet.dev-connection-ecs-fargate-vpn.id]
+    assign_public_ip = false
+    security_groups  = [aws_security_group.sg.id]
+  }
+}
+
+
+
+# Create PrivateLink to access ECR without internet connection
+resource "aws_vpc_endpoint" "dev-connection-ecs-fargate-vpn-s3" {
+  vpc_id            = aws_vpc.dev-connection-ecs-fargate-vpn.id
+  service_name      = "com.amazonaws.us-west-1.s3"
+  vpc_endpoint_type = "Gateway"
+  route_table_ids   = [aws_vpc.dev-connection-ecs-fargate-vpn.main_route_table_id]
+
+  tags = {
+    Name = "crafting-notebook-demo-s3"
+  }
+}
+
+resource "aws_vpc_endpoint" "dev-connection-ecs-fargate-vpn-ecr-dkr" {
+  vpc_id              = aws_vpc.dev-connection-ecs-fargate-vpn.id
+  service_name        = "com.amazonaws.us-west-1.ecr.dkr"
+  vpc_endpoint_type   = "Interface"
+  private_dns_enabled = true
+  subnet_ids          = [aws_subnet.dev-connection-ecs-fargate-vpn.id]
+  security_group_ids  = [aws_security_group.sg.id]
+
+  tags = {
+    Name = "crafting-notebook-demo-ecr-dkr"
+  }
+}
+
+resource "aws_vpc_endpoint" "dev-connection-ecs-fargate-vpn-ecr-api" {
+  vpc_id              = aws_vpc.dev-connection-ecs-fargate-vpn.id
+  service_name        = "com.amazonaws.us-west-1.ecr.api"
+  vpc_endpoint_type   = "Interface"
+  private_dns_enabled = true
+  subnet_ids          = [aws_subnet.dev-connection-ecs-fargate-vpn.id]
+  security_group_ids  = [aws_security_group.sg.id]
+
+  tags = {
+    Name = "crafting-notebook-demo-ecr-api"
+  }
+}
+
+resource "aws_vpc_endpoint" "dev-connection-ecs-fargate-vpn-logs" {
+  vpc_id              = aws_vpc.dev-connection-ecs-fargate-vpn.id
+  service_name        = "com.amazonaws.us-west-1.logs"
+  vpc_endpoint_type   = "Interface"
+  private_dns_enabled = true
+  subnet_ids          = [aws_subnet.dev-connection-ecs-fargate-vpn.id]
+  security_group_ids  = [aws_security_group.sg.id]
+
+  tags = {
+    Name = "crafting-notebook-demo-logs"
+  }
+}
+
+resource "aws_acm_certificate" "cert" {
+  private_key      = file(data.external.tls.result.server_key)
+  certificate_body = file(data.external.tls.result.server_cert)
+
+  certificate_chain = file(data.external.tls.result.ca_cert)
+}
```

Comment:
Please do not generate private key/cert in local file system. Use terraform `tls_...` resources. Example: https://github.com/crafting-dev/sandbox-deploy/tree/master/tf/modules/sandbox/aws/vpn